kani-compiler/src/kani_middle/transform/check_uninit/ty_layout.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+//! Utility functions that help calculate type layout.
+
+use stable_mir::abi::{FieldsShape, Scalar, TagEncoding, ValueAbi, VariantsShape};
+use stable_mir::target::{MachineInfo, MachineSize};
+use stable_mir::ty::{AdtKind, IndexedVal, RigidTy, Ty, TyKind, UintTy};
+use stable_mir::CrateDef;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+struct DataBytes {
+    /// Offset in bytes.
+    offset: usize,
+    /// Size of this data chunk.
+    size: MachineSize,
+}
+
+pub struct TypeLayout {
+    size_in_bytes: usize,
+    data_chunks: Vec<DataBytes>,
+}
+
+impl TypeLayout {
+    pub fn to_byte_mask(&self) -> Vec<bool> {
+        let mut layout_mask = vec![false; self.size_in_bytes];
+        for data_bytes in self.data_chunks.iter() {
+            for layout_item in
+                layout_mask.iter_mut().skip(data_bytes.offset).take(data_bytes.size.bytes())
+            {
+                *layout_item = true;
+            }
+        }
+        layout_mask
+    }
+}
+
+// Depending on whether the type is statically or dynamically sized,
+// the layout of the element or the layout of the actual type is returned.
+pub enum PointeeLayout {
+    Static { layout: TypeLayout },
```

Comment:
```suggestion
    /// Layout of sized objects
    Sized { layout: TypeLayout },
```
I think it makes sense to just use the Rust term for these things.